File: data_db/src/main/java/br/org/restaurantepopular/data_db/mapper/ProductTypeEntityMapper.java
package br.org.restaurantepopular.data_db.mapper;

import br.org.restaurantepopular.data_db.entity.ProductTypeEntity;
import br.org.restaurantepopular.entity.ProductType;

public class ProductTypeEntityMapper {

    public static ProductTypeEntity toProductTypeEntity(ProductType productType) {
        return new ProductTypeEntity.Builder()
                .id(productType.getId())
                .name(productType.getName())
                .build();
    }

    public static ProductType toProductType(ProductTypeEntity productTypeEntity) {
        return new ProductType.ProductTypeBuilder()
                .id(productTypeEntity.getId())
                .name(productTypeEntity.getName())
                .build();

    }
}

File: data_db/src/main/java/br/org/restaurantepopular/data_db/mapper/ProductTypeEntityMapper.kt
package br.org.restaurantepopular.data_db.mapper

import br.org.restaurantepopular.data_db.entity.ProductTypeEntity
import br.org.restaurantepopular.entity.ProductType

fun ProductType.toProductTypeEntity(): ProductTypeEntity {
    return ProductTypeEntity(
        id = this.id,
        name = this.name
    )
}

fun ProductTypeEntity.toProductType(): ProductType {
    return ProductType(
        id = this.id,
        name = this.name
    )

}